### PR TITLE
Filter tasks by user & create category

### PR DESCRIPTION
**Pull request by Pardiez**

### CODE REVIEW

File: api/main.rb [hunk 1]
```diff
@@ -171,18 +171,15 @@ def filter(params, missing = true)
           route_param :category_id do
             post "/" do
               person = Person.find_by_id(params[:id])
-              person.dislike_categories.delete_if { |cat| p cat.id; cat.id == params[:category_id].to_i }
-              if person.save
+              if person.addFavoriteCategory(params[:category_id])
```
Comment (esebastian) [inline api/main.rb]: the naming is not ruby-ish

File: api/main.rb [hunk 2]
```diff
@@ -171,18 +171,15 @@ def filter(params, missing = true)
...
               else
                 error_400(person)
               end
             end
             delete "/" do
               person = Person.find_by_id(params[:id])
-              category = Category.find_by_id(params[:category_id])
-              person.dislike_categories << category
-              if person.save
+              if person.removeFavoriteCategory(params[:category_id])
```
Comment (esebastian) [inline api/main.rb]: the naming is not ruby-ish

File: api/main.rb [hunk 3]
```diff
@@ -28,8 +28,10 @@ class API < Grape::API
     resource :tasks do
 
       get "/" do
-        return present Task.past_tasks, with: Entities::Task if params[:filter] == 'past'
-        present Task.future_tasks, with: Entities::Task
+        skip_categories = Person.getSkippedCategories(params[:user_id])
```
Comment (esebastian) [inline api/main.rb]: the naming is not ruby-ish

File: api/main.rb [hunk 4]
```diff
@@ -28,8 +28,10 @@ class API < Grape::API
     resource :tasks do
 
       get "/" do
-        return present Task.past_tasks, with: Entities::Task if params[:filter] == 'past'
-        present Task.future_tasks, with: Entities::Task
+        skip_categories = Person.get_skipped_categories(params[:user_id])
```
Comment (elmendalerenda) [inline api/main.rb:61]: me gusta skipped

File: api/models/task.rb [hunk 5]
```diff
@@ -59,12 +59,12 @@ def find_by_id(id)
         task
       end
 
-      def future_tasks
-        all(:active => true, :from_date.gte => Time.now, :order => [ :from_date.asc ])
+      def future_tasks(skip_categories)
```
Comment (elmendalerenda) [inline api/models/task.rb:62]: se merece un test ya no?

File: app/src/providers/task.service.ts [hunk 6]
```diff
@@ -30,13 +30,15 @@ export class TaskService {
     return task
   }
 
-  getFutureTasks(): Observable<Task[]> {
-    return this.http.get(`${this.huertaskApiUrl}/tasks/`)
+  getFutureTasks(user_id = null): Observable<Task[]> {
```
Comment (elmendalerenda) [inline app/src/providers/task.service.ts:33]: los metodos son iguales lo unico q cambia es la url

File: api/models/task.rb (hunk 5 above)
Comment (Pardiez) [reply api/models/task.rb:62]: Añadido.: https://github.com/devscola/huertask-app/pull/77/commits/9f4946f45624450e8dc013ac288182428cc6d3c5

Aún asi me siguen pareciendo tests demasiado crípticos,

File: app/src/providers/task.service.ts (hunk 6 above)
Comment (Pardiez) [reply app/src/providers/task.service.ts:33]: Extract method: https://github.com/devscola/huertask-app/pull/77/commits/3a41dd1bb269be298fc0d17b1431b05d94a75034

**Pull request merged**